src/connector/src/pulsar/source/reader.rs

```diff
@@ -170,15 +170,15 @@ impl SplitReader for PulsarSplitReader {
 
     async fn new(
         props: PulsarProperties,
-        state: ConnectorStateV2,
+        state: ConnectorState,
         _columns: Option<Vec<Column>>,
     ) -> Result<Self>
     where
         Self: Sized,
     {
         let (tx, rx) = mpsc::unbounded_channel();
         let rx = UnboundedReceiverStream::from(rx);
-        let splits = try_match_expand!(state, ConnectorStateV2::Splits).map_err(|e| anyhow!(e))?;
+        let splits = state.unwrap();
```

Comment:
When will state be None?

Reply:
it won't be None here. `ConnectorState.is_none()` is handled at https://github.com/singularity-data/risingwave/blob/2877c1fa64777a0cc26da1b6b0978c20619071c4/src/connector/src/macros.rs#L109, which will start a dummy split reader instead of a real one.